.github/workflows/check-python.yml

```diff
@@ -28,7 +28,7 @@ jobs:
         run: "pip install -r backend/compact-connect/requirements-dev.txt"
 
       - name: Lint Code
-        run: "cd backend; pylint $(git ls-files '*.py')"
+        run: "cd backend; ruff check $(git ls-files '*.py')"
```

Comment:
unless you intentionally want to _ignore_ .py files that are not part of the git repository, you should be able to run `ruff check` to recursively lint all python files.

(same comment for `format_python.sh`)

Reply:
I'm good either way. I can see the benefit of not checking files that are not added to the repo, as sometimes I have scratch files that I don't want the shell script to flag me on. But if a file is checked into the git repository, obviously we will want to check it. My personal vote is to keep this as it is, but I'm curious if @jusdino has a different opinion.

Reply:
I don't feel strongly either, but generally agree with @landonshumway-ia - the check is intended to be specific to files that are checked in. Of course, given that this is run only against the GitHub repo, freshly checked out by automation, I think the distinction is moot anyway.

Reply:
Sounds good. That's fine with me!